test/semantics/modfile23.f90

```diff
@@ -0,0 +1,184 @@
+! Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+!
+! Licensed under the Apache License, Version 2.0 (the "License");
+! you may not use this file except in compliance with the License.
+! You may obtain a copy of the License at
+!
+!     http://www.apache.org/licenses/LICENSE-2.0
+!
+! Unless required by applicable law or agreed to in writing, software
+! distributed under the License is distributed on an "AS IS" BASIS,
+! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+! See the License for the specific language governing permissions and
+! limitations under the License.
+
+! Test that subprogram interfaces get all of the symbols that they need.
+
+module m1
+  integer(8) :: i
+  type t1
+    sequence
```

Comment:
Is t1 a legal type ? From C740 in 7.5.2.3:

> If SEQUENCE appears, the type shall have at least one component [...]

Maybe it does not matter for what you are testing here though.

Reply:
It doesn't matter for this test, but I don't want the test to fail when we start checking that constraint, so I'll fix it. Thanks for catching it.